Sources/TensorFlow/Layers/Dropout.swift

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+fileprivate extension Tensor where Scalar: TensorFlowFloatingPoint {
+     /// Computes dropout given a probability.
+     // TODO: Remove the underscore once `droppingOut(probability:)` has been removed.
+    @differentiable(wrt: self where Scalar: Differentiable)
+    func _droppingOut(probability: Double) -> Tensor {
+        let noise = Tensor(randomUniform: shape)
+        let keepMask = noise .>= Scalar(probability)
+        let keepProbability = Scalar(1.0 - probability)
+        return self * Tensor(keepMask) / Tensor(keepProbability)
+    }
+}
+
+public extension Tensor where Scalar: TensorFlowFloatingPoint {
+    /// Computes dropout given a probability.
+    @available(*, deprecated, message: """
+        This API will be removed after Swift for TensorFlow 0.6.
+        For dropout, use the `Dropout` layer.
+        """)
+    @differentiable(wrt: self where Scalar: Differentiable)
+    func droppingOut(probability: Double) -> Tensor {
+        _droppingOut(probability: probability)
+    }
+}
+
+/// A dropout layer.
+///
+/// Dropout consists in randomly setting a fraction of input units to `0` at each update during
+/// training time, which helps prevent overfitting.
+@frozen
+public struct Dropout<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
+    @noDerivative public let probability: Double
+
+    /// Creates a dropout layer.
+    ///
+    /// - Parameter probability: The drop probability.
+    /// - Precondition: probability must be a value between 0 and 1 (inclusive). 
+    public init(probability: Double) {
+        precondition(0...1 ~= probability,
+            "Probability must be a value between 0 and 1 (inclusive) but is \(probability)")
```

Comment:
FWIW, although it's generally better to have one logical change per PR, given that this is just one statement, I think it's probably okay to leave in. 

Note: this change would break users who relied upon probability outside the range acting as if it's clamped to the range. I think this is okay, however, as this likely indicates a bug. That said, if we wanted to, we could provide a convenience initializer that would do the clamping to 0...1 and then delegate to this initializer. (This could be done in a follow-on PR.)

Reply:
> FWIW, although it's generally better to have one logical change per PR, given that this is just one statement, I think it's probably okay to leave in.

You are absolutely right. I forgot to exclude it in the commit--will look more closely in the future.

> Note: this change would break users who relied upon probability outside the range acting as if it's clamped to the range. I think this is okay, however, as this likely indicates a bug. That said, if we wanted to, we could provide a convenience initializer that would do the clamping to 0...1 and then delegate to this initializer. (This could be done in a follow-on PR.)

The convenience initializer ia a great idea. I think it should show some kind of message though indicating the input value will be clamped. Should this be a permanent or temporary initializer (with deprecation warning)?